apps/3d_rec_framework/CMakeLists.txt

```diff
@@ -13,7 +13,7 @@ else(NOT VTK_FOUND)
 endif(NOT VTK_FOUND)
 
 # OpenNI found?
-if(NOT OPENNI_FOUND OR NOT BUILD_OPENNI)
+if(NOT OPENNI_FOUND OR NOT WITH_OPENNI)
```

Comment:
To make it clear, I have in mind something like this:

```cmake
option(WITH_OPENNI "OpenNI driver support" ON)

if(WITH_OPENNI)
  find_package(OpenNI)
  if(NOT OPENNI_FOUND)
    set(WITH_OPENNI OFF
        CACHE BOOL "OpenNI driver support" FORCE)
    message(SEND_ERROR "Not building OpenNI driver support: OpenNI not found")
  else()
    include_directories(SYSTEM ${OPENNI_INCLUDE_DIRS})
  endif()
endif()
```

And later control other features based on `WITH_OPENNI` value.